processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -56,12 +69,14 @@
 import java.util.Collection;
 import java.util.Comparator;
 import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class InDimFilter implements DimFilter
+public class InDimFilter implements DimFilter, Filter
```

Comment:
Imagining a future world where `DimFilter` and `Filter` are combined, in the future can we rename this to `InFilter` instead of `InDimFilter` and drop all the `DimFilter` terminology? 

Is the place where the `DimFilter` implementations live `org.apache.druid.query.filter` more appropriate packaging than the place where the `Filter` implementations live `org.apache.druid.segment.filter`?

Reply:
> Imagining a future world where `DimFilter` and `Filter` are combined, in the future can we rename this to `InFilter` instead of `InDimFilter` and drop all the `DimFilter` terminology?

I think that would be best, yes.

> Is the place where the `DimFilter` implementations live `org.apache.druid.query.filter` more appropriate packaging than the place where the `Filter` implementations live `org.apache.druid.segment.filter`?

I think you could argue for either one. Just from looking at them, it seems like most other stuff that mixes Jackson API and implementation lives in `org.apache.druid.query`.